score-server/src/main/java/bio/overture/score/server/repository/s3/S3DownloadService.java

```diff
@@ -136,21 +112,56 @@ public ObjectSpecification download(
       } else {
         parts = partCalculator.divide(offset, length);
       }
+      if (objectSpec == null) {
+        ObjectMetadata metadata =
+            s3Client.getObjectMetadata(
+                bucketNamingService.getStateBucketName(objectId), objectKey.getKey());
+        fillPartUrls(objectKey, parts, false, forExternalUse);
+        objectSpec =
+            new ObjectSpecification(
+                objectKey.getKey(), objectId, objectId, parts, length, metadata.getETag(), false);
```

Comment:
We are not supposed to be using `length` from our function arguments to define the ObjectSpec, this value should be coming from reading the object information in S3.

On line 139 there is a comparison done between the objectSpec's object size and the `length` argument, so we can't define the objectSpec using the same argument... otherwise that check isn't useful.

Reply:
The ObjectSpecification is created using the metadata.getContentLength() which ensures the object size is accurate as per the S3 stored information.